exec/main.c

```diff
@@ -1183,15 +1184,35 @@ static int corosync_move_to_root_cgroup(void) {
 	 */
 	f = fopen("/sys/fs/cgroup/cpu/cpu.rt_runtime_us", "rt");
 	if (f == NULL) {
-		log_printf(LOGSYS_LEVEL_DEBUG, "cpu.rt_runtime_us doesn't exists -> "
-		    "system without cgroup or with disabled CONFIG_RT_GROUP_SCHED");
+		/*
+		 * Try cgroup v2
+		 */
+		f = fopen("/sys/fs/cgroup/cgroup.procs", "rt");
+		if (f == NULL) {
+			log_printf(LOG_DEBUG, "cpu.rt_runtime_us or cgroup.procs doesn't exists -> "
+			    "system without cgroup or with disabled CONFIG_RT_GROUP_SCHED");
 
-		res = 0;
-		goto exit_res;
+			res = 0;
+			goto exit_res;
+		} else {
+			cgroup_ver = 2;
+		}
+	} else {
+		cgroup_ver = 1;
 	}
 	(void)fclose(f);
 
-	f = fopen("/sys/fs/cgroup/cpu/tasks", "w");
+	assert(cgroup_ver == 1 || cgroup_ver == 2);
```

Comment:
I'm not convinced this should be an assert. Adding a default: to the switch below and a nice message would be more user-friendly IMHO

Reply:
Agree and it was exactly like that in previous version. But coverity is "clever" and found out, that there is no way how `cgroup_ver` can be other value than 1 and 2, so it reported error that default case is never reachable. Honestly, it is really bug in the code when `cgroup_ver` is unset so we may delete this case completelly, but I would rather let either compiler (ideally) or runtime error when cgroup_ver 3 is added and we forget to check the code... I will rework this to enum and remove the assert because then compiler starts complaining.